app/controllers/events_controller.rb

```diff
@@ -0,0 +1,48 @@
+class EventsController < ApplicationController
+  before_action :require_login, only: [:show]
+
+  def index
+    @events = Event.all
```

Comment:
- Add @past_events = Event.previous_events and @upcoming_events = Event.upcoming_events